### PR TITLE
[IMPROVEMENT] Add Documentation

### PR DESCRIPTION
**Pull request by TheFaR8**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Added Documentation (fulfill #21)

### CODE REVIEW

File: mod_deploy/controllers.py [hunk 1]
```diff
@@ -63,6 +63,9 @@ def decorated_function(*args, **kwargs):
 
 
 def is_valid_signature(x_hub_signature, data, private_key):
+    """
+    Check github deploy signature key (valid or not)
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: I'm missing the passed in parameters with their types.

Also, if you add documentation for this file, please do it for the entire file.

File: mod_sample/controllers.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def display_sample_info(sample):
     status_release = 'Unknown'
     if len(regression_tests) > 0:
         if test_commit is not None:
+            # Test the commit based on the Regression Test
```
Comment (canihavesomecoffee) [inline mod_sample/controllers.py]: Also, if you add documentation for this file, please do it for the entire file.

**Comment by TheFaR8** [comment]
I have to do 3 doc-blocks or 3 files?
thanks

**Comment by canihavesomecoffee** [comment]
The task is different from the issue you referenced, that's why I was confused.

3 doc-blocks, but full ones that meet the PEP requirements. The 2 single line comments do not count.

**Comment by TheFaR8** [comment]
Ahh
I must be misread the task

File: mod_deploy/controllers.py [hunk 3]
```diff
@@ -63,6 +63,9 @@ def decorated_function(*args, **kwargs):
 
 
 def is_valid_signature(x_hub_signature, data, private_key):
+    """
+    Re-check if the GitHub hook request got valid signature
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: Please make sure the doc-blocks fullfil the PEP-257, as listed here: https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md#requirements-for-a-pull-request

Comment (TheFaR8) [reply mod_deploy/controllers.py]: for all doc-blocks right?

Comment (canihavesomecoffee) [reply mod_deploy/controllers.py]: Besides this, please also add descriptions for the types and arguments (see https://github.com/CCExtractor/sample-platform/blob/master/config_parser.py of an example)

Comment (canihavesomecoffee) [reply mod_deploy/controllers.py]: Yes, all added doc-blocks need to conform to that PEP.

File: mod_sample/controllers.py [hunk 4]
```diff
@@ -42,7 +42,7 @@ class SampleNotFoundException(Exception):
 
     def __init__(self, message):
         Exception.__init__(self)
-        self.message = message
+        self.= message
```
Comment (canihavesomecoffee) [inline mod_sample/controllers.py]: This won't work. Did you use an IDE for this?

**Comment by TheFaR8** [comment]
Seems like too many commit made :\

Will re-do them in one commit tomorrow

File: mod_sample/controllers.py (hunk 4 above)
Comment (TheFaR8) [reply mod_sample/controllers.py]: Typo

File: mod_deploy/controllers.py [hunk 5]
```diff
@@ -72,6 +81,12 @@ def is_valid_signature(x_hub_signature, data, private_key):
 @mod_deploy.route('/deploy', methods=['GET', 'POST'])
 @request_from_github()
 def deploy():
+    """
+    Deploy the GitHub request to the test platform.
+    
+    :request, abort_code, g: library imported by flask
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: This is not the correct and wanted type of annotation.

Only arguments of the function (`:param` and the type of that parm `:type`) should be added. Please take a look at the example I linked yesterday.

File: mod_sample/controllers.py [hunk 6]
```diff
@@ -133,6 +133,12 @@ def display_sample_info(sample):
 @mod_sample.errorhandler(SampleNotFoundException)
 @template_renderer('sample/sample_not_found.html', 404)
 def not_found(error):
+    """
+    Pretty much self explaintory.
```
Comment (canihavesomecoffee) [inline mod_sample/controllers.py]: If it's "pretty much self explanatory", why add that comment at all?

File: mod_deploy/controllers.py (hunk 5 above)
Comment (TheFaR8) [reply mod_deploy/controllers.py]: No need to add everything?
only the important one?

Comment (canihavesomecoffee) [reply mod_deploy/controllers.py]: Deploy has no arguments, so besides the description nothing is needed here.

**Comment by TheFaR8** [comment]
Updated
Thanks

**Comment by TheFaR8** [comment]
Updated
Thanks

File: mod_sample/controllers.py [hunk 7]
```diff
@@ -186,6 +186,12 @@ def serve_file_download(file_name, sub_folder='',
 
 @mod_sample.route('/download/<sample_id>')
 def download_sample(sample_id):
+    """
+    Making sure that the sample exist (so it can be downloaded).
+
+    :param sample_id: numbers
+    :return: error
```
Comment (canihavesomecoffee) [inline mod_sample/controllers.py]: That is incorrect.

**Comment by canihavesomecoffee** [comment]
Please also rebase against the latest master version

File: mod_sample/controllers.py (hunk 7 above)
Comment (TheFaR8) [reply mod_sample/controllers.py]: Done

File: mod_deploy/controllers.py [hunk 8]
```diff
@@ -71,6 +78,7 @@ def is_valid_signature(x_hub_signature, data, private_key):
 @mod_deploy.route('/deploy', methods=['GET', 'POST'])
 @request_from_github()
 def deploy():
+    """Deploy the GitHub request to the test platform."""
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: Please put the comment on a separate line, like the others.

File: mod_sample/controllers.py [hunk 9]
```diff
@@ -172,6 +172,13 @@ def serve_file_download(file_name, sub_folder='', content_type='application/octe
 
 @mod_sample.route('/download/<sample_id>')
 def download_sample(sample_id):
+    """
+    Download sample if the sample exist
+
+    :param sample_id: numbers
+    :raise: error
+    :return: function above
```
Comment (canihavesomecoffee) [inline mod_sample/controllers.py]: That's still incorrect, and totally useless if someone decides to add a function in between.

Comment (TheFaR8) [reply mod_sample/controllers.py]: I think it is better to delete these doc-blocks
as I rethink that this doc-blocks are self explaintory

**Pull request merged**